sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -728,9 +728,15 @@ class DataFrameAggregateSuite extends QueryTest with SharedSQLContext {
     import java.lang.Float.floatToRawIntBits
     import java.lang.Double.doubleToRawLongBits
 
-    // 0.0/0.0 and NaN are different values.
-    assert(floatToRawIntBits(0.0f/0.0f) != floatToRawIntBits(Float.NaN))
```

Comment:
This is to prove that these special float values have different binary representation, which would break the unsafe row format if Spark doesn't take care of it.

Reply:
More specifically, NaN has different binary representations in JVM. Is this changed in the aarch64 platform?

Reply:
It sounds like this is the case, but, if several NaN representations that are _different_ on x86 are the _same_ on ARM then it should still work OK. Let me run tests to figure it out.

So, I guess the question is, do we test whatever would break if Spark doesn't handle this? if so then we don't necessarily have to make this specific assertion is true, because it isn't true in all cases according to the JDK.

Reply:
I have one question. Does this difference between two architectures cause the difference of results in a Spark program? Could you please add such a test case?

If this difference does not cause any difference of results, we do not need to take care of this difference. In other words, I think that we do not need to add this test.

Reply:
Interesting.